File: app/src/main/java/com/cliff/conch/box/scene/Singleton.java
package com.cliff.conch.box.scene;

import com.cliff.conch.box.reflect.RefClass;
import com.cliff.conch.box.reflect.RefMethod;
import com.cliff.conch.box.reflect.RefObject;

public class Singleton {
    public static Class<?> TYPE = RefClass.load(Singleton.class, "android.util.Singleton");
    public static RefMethod<?> get;
    public static RefObject<Object> mInstance;
}

File: reflect-android/src/main/java/reflect/android/util/Singleton.kt
package reflect.android.util

import com.cliff.reflection.common.IReflect
import com.cliff.reflection.common.annotation.PField
import com.cliff.reflection.common.annotation.PMethod
import com.cliff.reflection.common.annotation.ProxyClass

@ProxyClass("android.util.Singleton")
interface Singleton : IReflect {
    companion object

    @PField
    var mInstance: Any?

    @PMethod
    fun get(): Any?
}